src/braket/parametric/free_parameter_expression.py

```diff
@@ -51,8 +69,11 @@ def __init__(self, expression: Union[FreeParameterExpression, Number, Expr, str]
             ast.Pow: self.__pow__,
             ast.USub: self.__neg__,
         }
+        self._type = _type if _type is not None else FloatType()
```

Comment:
We can validate the type is a valid type if it is in https://github.com/amazon-braket/amazon-braket-default-simulator-python/blob/1044c34ebf09c1a4c73a2b0e5ecd75f3877077fa/src/braket/default_simulator/openqasm/parser/openqasm_ast.py#L21C1-L21C8